scripts/gen-os-secrets.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+if [ $# -ne 1 ]; then
+    echo "$(basename "$0") <output-file>" >&2
+    exit 1
+fi
+
+set -o pipefail
+
+if ! type -p yq > /dev/null; then
+    echo "You must have yq installed to use this script" >&2
+    exit 1
+fi
+
+if ! type -p kubectl > /dev/null; then
+    echo "You must have kubectl installed to use this script" >&2
+    exit 1
+fi
+
```

Comment:
```suggestion
if ! type -p kubectl > /dev/null; then
    echo "You must have kubectl installed to use this script" >&2
    exit 1
fi

KUSTOMIZE_VERSION=$(kubectl version --client -o yaml | yq .kustomizeVersion)
if ! (echo -e "v5.0.0\n$KUSTOMIZE_VERSION" | sort -V -C); then
  echo "kustomize needs to be at version 5.0.0 or newer (comes with kubectl 1.27+)"
  exit 1
fi

```

`additionalValuesFiles` has been [added](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.0.0) in v5.0.0 so we should probably check for availability of this or newer version